### PR TITLE
DSG-2861 fix: fixed base route so activeTabIndex updates correctly

### PR DESCRIPTION
**Pull request by benjaminlu-nuv**
# Basics

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Fix wrong tab highlighted on refresh

## Screenshot of the affected changes
https://www.loom.com/share/d91adfbe0fee4d7896ec29a5e5c021c0

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

**Review by lucywang-nv — APPROVED** [review]
Looks good!

File: libs/employer/feature/dashboard/src/lib/components/employer-detail/employer-detail.component.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
   INavigableTab,
   INuverialBreadCrumb,
   INuverialSelectOption,
+  LoadingService,
```
Comment (davidyoung77) [inline libs/employer/feature/dashboard/src/lib/components/employer-detail/employer-detail.component.ts:12]: Please remember to remove the spinner component from imports, not sure we have been doing this

**Review by davidyoung77 — APPROVED** [review]
Nice, thanks for updating the loading also

File: libs/employer/feature/dashboard/src/lib/components/employer-detail/employer-detail.component.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ export class EmployerDetailComponent implements OnDestroy {
         return EMPTY;
       }
 
-      this.baseRoute = `/dashboard/${this._categoryRoute}`;
+      this.baseRoute = `/dashboard/${this._categoryRoute}/transaction/${transactionId}`;
```
Comment (thechechogomez) [inline libs/employer/feature/dashboard/src/lib/components/employer-detail/employer-detail.component.ts:124]: @benjaminlu-nuv Let's double check this as I think this base route should only contain transaction if there is indeed a transaction id

Comment (thechechogomez) [reply libs/employer/feature/dashboard/src/lib/components/employer-detail/employer-detail.component.ts:124]: https://github.com/Nuvalence/dsgov-web/assets/60438413/be616e05-36f3-4a26-88e9-080019869575

Yeah, it indeed breaks the employer detail when it doesn't contain a transaction i.

**Pull request merged**